site/app/models/GradingOrder.php

```diff
@@ -496,18 +496,31 @@ public function getSortedGradedGradeables() {
         );
         $gg_idx = [];
         $unsorted = [];
+        $items = [];
         foreach ($iter as $gg) {
-            $idx = $this->getSubmitterIndex($gg->getSubmitter());
-            //Should never happen, but better to be safe
-            if ($idx === false) {
-                $unsorted[] = $gg;
-            }
-            else {
-                $gg_idx[$idx] = $gg;
+            $items[$gg->getSubmitter()->getId()] = $gg;
+        }
+        $count = 0;
+
+        //Iterate through all sections and their submitters to find this one
+        foreach ($this->section_submitters as $name => $section) {
```

Comment:
I'm a little confused about what this code is trying to do.  Is there a particular reason why it's being done this way?  From my naive point of view, not entirely understanding the code but just looking at the structure of it, it seems odd that the best way to do this would be to look things up in linear time repeatedly with a double-nested loop structure.